tests/data/tensorflow_mnist/mnist.py

```diff
@@ -0,0 +1,195 @@
+    x_train = np.load(os.path.join(base_dir, 'train_data.npy'))
+    y_train = np.load(os.path.join(base_dir, 'train_labels.npy'))
+    return x_train, y_train
+
+def _load_testing_data(base_dir):
+    x_test = np.load(os.path.join(base_dir, 'eval_data.npy'))
+    y_test = np.load(os.path.join(base_dir, 'eval_labels.npy'))
+    return x_test, y_test
+
+def _parse_args():
+
+    parser = argparse.ArgumentParser()
+
+    # hyperparameters sent by the client are passed as command-line arguments to the script.
+    parser.add_argument('--epochs', type=int, default=1)
+    # Data, model, and output directories
+    parser.add_argument('--output-data-dir', type=str, default=os.environ['SM_OUTPUT_DATA_DIR'])
+    parser.add_argument('--model_dir', type=str, default=os.environ['SM_MODEL_DIR'])
+    parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAINING'])
+    parser.add_argument('--hosts', type=list, default=json.loads(os.environ['SM_HOSTS']))
+    parser.add_argument('--current-host', type=str, default=os.environ['SM_CURRENT_HOST'])
+
+    return parser.parse_known_args()
+
+def serving_input_fn():
+    inputs = {'x': tf.placeholder(tf.float32, [None, 784])}
+    return tf.estimator.export.ServingInputReceiver(inputs, inputs)
+
+if __name__ == "__main__":
+    args, unknown = _parse_args()
+    tf.logging.set_verbosity(tf.logging.DEBUG)
+    _handler = _logging.StreamHandler(_sys.stdout)
+    tf_logger = tf_logging._get_logger()
+    tf_logger.handlers = [_handler]
+
+    if args.model_dir.startswith('s3://'):
+        os.environ['S3_REGION'] = 'us-west-2'
+        os.environ['TF_CPP_MIN_LOG_LEVEL'] = '1'
+        os.environ['S3_USE_HTTPS'] = '1'
+
+    train_data, train_labels = _load_training_data(args.train)
+    eval_data, eval_labels = _load_testing_data(args.train)
+
+    # Create the Estimator
+    mnist_classifier = tf.estimator.Estimator(
+        model_fn=cnn_model_fn, model_dir=args.model_dir)
+
+    # Set up logging for predictions
+    # Log the values in the "Softmax" tensor with label "probabilities"
+    tensors_to_log = {"probabilities": "softmax_tensor"}
+    logging_hook = tf.train.LoggingTensorHook(
+        tensors=tensors_to_log, every_n_iter=50)
+
+    # Train the model
+    train_input_fn = tf.estimator.inputs.numpy_input_fn(
+        x={"x": train_data},
+        y=train_labels,
+        batch_size=100,
+        num_epochs=None,
+        shuffle=True)
+
+    # Evaluate the model and print results
+    eval_input_fn = tf.estimator.inputs.numpy_input_fn(
+        x={"x": eval_data},
+        y=eval_labels,
+        num_epochs=1,
+        shuffle=False)
+
+    train_spec = tf.estimator.TrainSpec(train_input_fn, max_steps=1000)
+    eval_spec = tf.estimator.EvalSpec(eval_input_fn)
+    tf.estimator.train_and_evaluate(mnist_classifier, train_spec, eval_spec)
+
+    if args.current_host == args.hosts[0]:
+        mnist_classifier.export_savedmodel(args.model_dir, serving_input_fn)
```

Comment:
Did you try:

```python 
feature_spec = {'x': tf.placeholder(tf.float32, [None, 784])}
serving_input_fn  = tf.estimator.export.build_parsing_serving_input_receiver_fn(feature_spec)
```

```

Reply:
I got invalid feature with this. If you don't mind I will leave this as is and fix it later.